src/main/java/org/broadinstitute/hellbender/utils/codecs/gtf/AbstractGtfCodec.java

```diff
@@ -474,26 +474,53 @@ boolean checkHeaderLineStartsWith(final List<String> header, final int lineNum,
 
     /**
      * Checks that the given header line number starts with the given text.
+     * Uses {@link #getAllLineComments()} to check for comments at the line starts.
      * @param header A {@link List<String>} containing a header to validate.
      * @param lineNum Line number in the header to check.
      * @param startingText {@link String} containing text that the line should start with
      * @param throwIfInvalid If {@code true} will throw a {@link UserException} instead of returning false.
      * @return {@code true} IFF the header line number {@code lineNum} starts with {@code startingText}; {@code false} otherwise.
      */
     boolean checkHeaderLineStartsWith(final List<String> header, final int lineNum, final String startingText, final boolean throwIfInvalid ) {
-        if ( !header.get(lineNum).startsWith(getLineComment() + startingText) ) {
+        boolean foundGoodLine = false;
+        for ( final String commentPrefix : getAllLineComments() ) {
```

Comment:
Seems like some duplicated code with isLineCommented but maybe complexity of sharing it makes it not worthwhile to refactor.

Reply:
Yeah - I was thinking about this as I was implementing it.  I can refactor it.